numpy/core/tests/test_multiarray.py

```diff
@@ -2787,7 +2787,7 @@ def test_dot_out_mem_overlap(self):
         # Test BLAS and non-BLAS code paths, including all dtypes
         # that dot() supports
         dtypes = [np.dtype(code) for code in np.typecodes['All']
-                  if code not in 'USVM']
+                  if code not in 'USVMm']
```

Comment:
a reviewer should probably confirm that the need for this with the casting changes makes sense; I think the need for this only snuck in after I prohibited bool from casting to timedelta64

Reply:
Calling dot on timedeltas is nonsense, so this change seems fine